packages/data-service/src/connection-secrets.ts

```diff
@@ -9,12 +9,9 @@ export interface ConnectionSecrets {
   sshTunnelPassphrase?: string;
   awsSessionToken?: string;
   tlsCertificateKeyFilePassword?: string;
+  proxyPassword?: string;
 }
 
-const AWS_SESSION_TOKEN_PROPERTY = 'AWS_SESSION_TOKEN';
-const AUTH_MECHANISM_PROPERTIES_PARAM = 'authMechanismProperties';
-const TLS_CERTIFICATE_KEY_FILE_PASSWORD_PARAM = 'tlsCertificateKeyFilePassword';
-
```

Comment:
Can we maybe establish a rule and put it to compass docs on how to use constants and strings? I have seen multiple times already how different developers several times change such places back and force 😄

Reply:
Yeah, so … I don’t feel super strongly about this, but `const FOO_PARAM = 'FOO'` does feel to me like it doesn’t provide a lot of value – usually, proper typescript definitions are the best way to both avoid typos and redundancy. I’m not sure how easy that would be for connection string options, though.

Reply:
It looks like it’s fairly doable to implement: https://github.com/mongodb-js/mongodb-connection-string-url/pull/14

We could use `uri.typedSearchParams<MongoClientOptions>()` instead of `uri.searchParams` in this case, it’s a bit more verbose, but it would ensure that we only handle options here that the driver understands

I wouldn’t hold up merging this PR over that though :)

Reply:
Definitely not a blocker!